docs/dev/opensearch-same-table-join-query.md

```diff
@@ -0,0 +1,1252 @@
+language is a relational language that provides the JOIN feature to relate tables together by field or expression.  OpenSearch
+offers a similar relation mapping within an index, that allows users to relate same-index documents using a parent-child
+relation.
+
+This proposal is to outline the limits of a same-table (ie same-index) join relation from SQL grammar, and provide use
+cases for same-table JOINs.
+
+# Table of Contents
+
+`TODO`
+
+# Same-Table Joins Query Design
+
+`TODO`: Overview of problem and what JOIN relations are aiming to solve
+
+## 1. Motivation
+
+`TODO`: Use cases, and SQL statements
+
+## 2. References
+
+* [[FEATURE] Support for subqueries or chaining of queries](https://github.com/opensearch-project/sql/issues/1441)
+* [[BUG] implicit join returns wrong results](https://github.com/opensearch-project/sql/issues/683)
+* [[FEATURE] Allow individual shards to be targeted during query execution](https://github.com/opensearch-project/sql/issues/1478)
+  * [Support for _routing field in SQL queries](https://github.com/opendistro-for-elasticsearch/sql/issues/1151)
+* [Nested Design](https://github.com/Bit-Quill/opensearch-project-sql/pull/262)
+* [opensearch.org: Join field type](https://opensearch.org/docs/latest/field-types/join/)
+* [PariQL](https://partiql.org/)
+
+## 3. Assumptions/Dependencies
+
+* _shard must be defined for any query using a same-table join
+* Same-Table joins requires that the following work is complete:
+    * Table/Indices can be aliased, for example: `SELECT t.fieldA, t.fieldB FROM table t WHERE t.fieldA = t.fieldB`
+
+### 3.1 Out of Scope
+
+* Fix for implicit JOINs [[BUG] implicit join returns wrong results](https://github.com/opensearch-project/sql/issues/683)
+* Fix for _shard or _routing [[FEATURE] Allow individual shards to be targeted during query execution](https://github.com/opensearch-project/sql/issues/1478)
+* Performance on the OpenSearch side
+* Multiple-shard queries using `join`
+* `join` using lateral-table support [PartiQL support in OpenSearch SQL V2](https://github.com/opensearch-project/sql/issues/1104)
+* Pagination using `join` or other complex types
+
+### 3.2 Design Decisions
+
+* JOIN-relations architecture is built to work in combination with the `nested` and `object` data types.  This allows enables
+  both same-table joins and nested object (using PartiQL) to share execution paths and syntax.  This also enables a shared
+  solution to expand support for `lateral` tables and enable `lateral join` and `lateral nested` queries.
+    * see [Add support for PArtiQL in V2 engine](https://github.com/opensearch-project/sql/issues/1104)
+
+### 3.3. Storage Type
+
+* `reference to opensearch doc`
+* `Describe how to define the JOIN-relation`
+* `What the datatype can do`
+* `What the datatype cannot do`
+
+## 4. Use Cases
+
+Use Cases have examples that is defined in the Annex.
```

Comment:
Maybe separate use cases and examples?

Reply:
Sorry.  This should say use cases with examples.